nhais-adaptor/edifact/incoming/parser/deserialiser.py

```diff
@@ -0,0 +1,100 @@
+import edifact.incoming.parser.creators as creators
+from edifact.incoming.models.message import MessageSegment
+from edifact.incoming.models.interchange import Interchange
+
+INTERCHANGE_HEADER_KEY = "UNB"
+MESSAGE_HEADER_KEY = "UNH"
+MESSAGE_BEGINNING_KEY = "BGM"
+MESSAGE_REGISTRATION_KEY = "S01"
+MESSAGE_PATIENT_KEY = "S02"
+MESSAGE_TRAILER_KEY = "UNT"
+INTERCHANGE_TRAILER_KEY = "UNZ"
+
+
+def extract_relevant_lines(original_dict, starting_pos, trigger_key):
+    """
+    From the original dict generate a smaller dict just containing the relevant lines based upon the trigger key
+    will keep looping till the terminating key is found in the terminating config
+    :param original_dict: The original larger dictionary
+    :param starting_pos: The starting position to start the loop from.
+    This is to prevent starting the loop from the start each time and be slightly more efficient
+    :param trigger_key: The trigger key for this section that will be used to find what the
+    terminating key for the section is
+    :return: A smaller dictionary with just the relevant lines for the section
+    """
+    terminating_config = {
+        INTERCHANGE_HEADER_KEY: [MESSAGE_HEADER_KEY],
+        MESSAGE_BEGINNING_KEY: [MESSAGE_REGISTRATION_KEY],
+        MESSAGE_REGISTRATION_KEY: [MESSAGE_PATIENT_KEY, MESSAGE_TRAILER_KEY],
+        MESSAGE_PATIENT_KEY: [MESSAGE_TRAILER_KEY]
+    }
+
+    new_dict = []
+    for (key, value) in original_dict[starting_pos:]:
+        if key not in terminating_config[trigger_key]:
+            new_dict.append((key, value))
+        else:
+            break
+
+    return new_dict
+
+
+def convert_to_dict(lines):

```

Comment:
As in other places, this isn't producing a dictionary, is it? It's returning a list of tuples.

Reply:
Yes it's a special type of dict which has now a specific `EdifactDict` type. Does calling these a `dict` still apply? It's still a key value pair but with potential duplicate keys

Reply:
I think now you've defined a specific type of edifact dictionary it's probably reasonable to stick with the dict name.